youtube_dl/extractor/mnet.py

```diff
@@ -0,0 +1,72 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import parse_duration
+
+
+class MnetIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?mnet\.(?:com|interest\.me)/tv/vod/(?:.*?\bclip_id=)?(?P<id>[0-9]+)'
+    _TESTS = [
+        {
+            'url': 'http://www.mnet.com/tv/vod/171008',
+            'md5': '6abd7a837fa9fe56d22709a60b19bffb',
+            'info_dict': {
+                'id': '171008',
+                'ext': 'flv',
+                'title': 'SS_이해인@히든박스',
+                'thumbnail': 're:^https?://.*\.jpg$',
+                'duration': 88,
+            },
+        },
+        {
+            'url': 'http://mnet.interest.me/tv/vod/172790',
+            'md5': '1552891a585b9f640a64e4ba89d51f6e',
+            'info_dict': {
+                'id': '172790',
+                'ext': 'flv',
+                'title': '[직캠]일대일아이컨택ㅣ최유정 -  ♬Yum-Yum(얌얌) @ 콘셉트 평가',
+                'thumbnail': 're:^https?://.*\.jpg$',
+                'duration': 209,
+            },
+        },
+        {
+            'url': 'http://www.mnet.com/tv/vod/vod_view.asp?clip_id=172790&tabMenu=',
+            'md5': '1552891a585b9f640a64e4ba89d51f6e',
+            'info_dict': {
+                'id': '172790',
+                'ext': 'flv',
+                'title': '[직캠]일대일아이컨택ㅣ최유정 -  ♬Yum-Yum(얌얌) @ 콘셉트 평가',
+                'thumbnail': 're:^https?://.*\.jpg$',
+                'duration': 209,
+            },
+        },
+    ]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        info_url = 'http://content.api.mnet.com/player/vodConfig?id=%s' % video_id
+        info = self._download_json(info_url, video_id)
+        info = info['data']['info']
+
+        title = info['title']
+        duration = parse_duration(info.get('time'))
+        try:
+            thumbnail = info['cover']['maxres']['url']
+        except KeyError:
+            thumbnail = None
```

Comment:
We usually use default fallbacks instead for shorter code: `info.get('cover', {}).get('maxres', {}).get('url')`